services/src/main/scala/cromwell/services/keyvalue/impl/KeyValueServiceActor.scala

```diff
@@ -30,23 +31,13 @@ case class KeyValueServiceActor(serviceConfig: Config, globalConfig: Config) ext
     }
   }
 
-  private def doPut(put: KvPut): Future[KvResponse] = {
-    upsertExecutionInfo(
-      put.pair.key.workflowId,
-      put.pair.key.jobKey,
-      Map(put.pair.key.key -> put.pair.value)
-    ).map(_ => KvPutSuccess(put))
+  private def doPut(put: KvPut): Future[KvResponse] = Future.successful {
```

Comment:
Perhaps, since this is currently CR~~UD~~. But things change and I'm very wary of "it's fine to have race conditions because they're unlikely"